File: src/main/java/itmo/is/project/repository/user/SpaceshipRepository.java
package itmo.is.project.repository.user;

import itmo.is.project.model.user.Spaceship;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface SpaceshipRepository extends JpaRepository<Spaceship, Integer> {
    Optional<Spaceship> findByPilotId(Integer pilotId);

    Optional<Spaceship> findByPilotUsername(String username);
}

File: src/main/kotlin/itmo/isproject/repository/user/SpaceshipRepository.kt
package itmo.isproject.repository.user

import itmo.isproject.model.user.Spaceship
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface SpaceshipRepository : JpaRepository<Spaceship, Int> {

    fun findByPilotId(pilotId: Int?): Spaceship?

    fun findByPilotUsernameInternal(username: String?): Spaceship?
}
